### PR TITLE
Nuke the env provider

### PR DESCRIPTION
**Pull request by glibsm**
The other side of #384

Env provider is being split up into two things:
1. Ability to interpolate env variables directly into YAML 
2. A small wrapper function around `os.Getenv` which will return `config.Value` to leverage the existing config infrastructure

### CODE REVIEW

**Comment by glibsm** [comment]
Side note: this will have been a lot easier to rip out if env specific tests were in `env_test.go` as opposed to all lumped into `config.go` :(

File: config/config.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ var (
 	_setupMux sync.Mutex
```
Comment (alsamylkin) [inline config/config.go]: Could you remove all this appRoot, environment vars above?

File: config/config.go [hunk 2]
```diff
@@ -120,13 +120,6 @@ func YamlProvider() ProviderFunc {
...
 
-// EnvProvider returns function to create environment based config provider
-func EnvProvider() ProviderFunc {
-	return func() (Provider, error) {
-		return NewEnvProvider(defaultEnvPrefix, nil), nil
-	}
-}
-
 // Environment returns current environment setup for the service
 func Environment() string {
```
Comment (alsamylkin) [inline config/config.go]: This should be removed as well? Also Path(), EnvironmentPrefix, Key etc..

Comment (glibsm) [inline config/config.go]: I thought unused functions were a compiler warning. Let me take a sweep and see what's left lingering

Comment (glibsm) [inline config/config.go]: - `Environment()` is used for something else
- `Path()` is used for YAML file loading (not affected by this)
- `EnvironmentPrefix` is also used as part of YAML loading

Can't remove environment stuff as part of the YAML loading :) 
Once we create the env package, I expect some of that might change.

**Comment by alsamylkin** [comment]
Changelog missing?

**Pull request merged**